platform/mellanox/mlnx-platform-api/sonic_platform/component.py

```diff
@@ -0,0 +1,199 @@
+#!/usr/bin/env python
+
+#############################################################################
+# Mellanox
+#
+# implementation of new platform api
+#############################################################################
+
+try:
+    from sonic_platform_base.component_base import ComponentBase
+    from glob import glob
+    import subprocess
+    import io
+    import re
+except ImportError as e:
+    raise ImportError(str(e) + "- required module not found")
+
+#components definitions
+COMPONENT_BIOS = "BIOS"
+COMPONENT_FIRMWARE = "ASIC-FIRMWARE"
+COMPONENT_CPLD = "CPLD"
+
+FW_QUERY_VERSION_COMMAND = 'mlxfwmanager --query'
+BIOS_QUERY_VERSION_COMMAND = 'dmidecode -t 11'
+CPLD_VERSION_FILE_PATTERN = '/var/run/hw-management/system/cpld?_version'
+CPLD_VERSION_MAX_LENGTH = 4
```

Comment:
@stephenxs what if platform has 3 or 2 CPLDs? How you will distinguish?

Reply:
This is the max length of version file. Currently the max version length is two which means it contains 2 digit at last. Anyway I think using four digits to represent the version number is enough.
It's not the number of CPLDs.

Reply:
@stephenxs ok. Thanks.